src/main/scala/beam/replanning/utilitybased/ChainBasedTourVehicleAllocator.scala

```diff
@@ -202,8 +206,10 @@ object ChainBasedTourVehicleAllocator {
     def apply(possibleVehicles: Vector[VehicleRecord], subtour: Subtour): SubtourRecord = {
       val trips = JavaConverters.collectionAsScalaIterable(subtour.getTrips)
       val startTime = Try {
-        trips.head.getOriginActivity.getStartTime
-      }.getOrElse(throw new RuntimeException(s"No endTime in ${trips.head.getOriginActivity}"))
+        trips.headOption.map(_.getOriginActivity.getStartTime).get
```

Comment:
Doesn't this trigger an OptionGet warning? How about:

```
val tripOriginActivityOption = trips.headOption.map(_.getOriginActivity)
val startTime = (
  Try{ tripOriginActivityOption.map(_.getStartTime) }
).toOption.flatten
.getOrElse(throw new RuntimeException(s"No startTime in $tripOriginActivityOption"))
```

This resolves all of the possible safety issues. I'm only unsure if `getOriginActivity` needs wrapped in the `Try`?

Reply:
The `Try` is redundant. After changes became as follows:
```Scala
val tripOriginActivityOption = trips.headOption.map(_.getOriginActivity)
val startTime = tripOriginActivityOption.map(_.getStartTime) 
  .getOrElse(throw new RuntimeException(s"No startTime in $tripOriginActivityOption"))
```